### ISSUE TITLE
Upload speeds won't top 20Mbit/s (~2.5MB/s)

### ISSUE THREAD

**Issue by colin19889** [JonathanTreffler/backblaze-personal-wine-container#154]
I installed this today. Docker running in a Proxmox LXC. Waited for the initial sync to finish and dropped in a 10GB file. Router (PfSense) is reporting 20Mbit/s upload (and the time it took to complete the upload matches that). I have a 3Gig symmetrical connection and the Proxmox server is on a 2.5Gig switch. I built a Speedtest Container in the same LXC just to make sure the problem wasn't on my side, and it Speedtests at 2.5Gig each way. Threads were set to 30 and then 100 (pausing/rebooting in between).  `docker stats` tells me  the container is using ~20% CPU and about 2GB RAM (I gave it 4GB). The drive I'm syncing is mounted to /mnt/Temp in the LXC and shows up as drive "D" in the container. 

Any thoughts? I'm on the Backblaze 15-day trial but I'm guessing that wouldn't make a difference.

**Comment by JonathanTreffler** [comment]
I have no magical solution, but I have some debugging help:
Big files (like your 10 GB file) get split into smaller files/chunks on disk, the hash is calculated and then the chunks are uploaded. If the media backing your container layer is slow this could lead to slow upload speeds.
In general this container is expected to be slower, than a native installation, because of the overhead of wine, but more than 20 mbit is definitely possible.
Trial should not make a difference.
A debugging step you could try is switching to a different container tag to :ubuntu22 (or :ubuntu18), the weirder bug reports here often get solved with random things like that (and nobody can explain why :laughing: )

**Comment by colin19889** [comment]
Thanks for the quick response. So tried a couple things to eliminate local issues.

1. Spun up a Windows VM, passed the same SSD I was using with Wine through and did a backup. Average 400-500Mbit/s, so I don't think it's the storage media causing the slowdown (I also used the same media for temps). Maybe it's the docker config? But I haven't ever had a problem like that before.
2. Tried both ubuntu22 and ubuntu18, speed will sometimes hit 30Mbit/s but nothing higher.

Any more suggestions?
